tensorflow/core/lib/strings/str_util.cc

```diff
@@ -416,6 +417,119 @@ bool ConsumeNonWhitespace(StringPiece* s, StringPiece* val) {
   }
 }
 
+Status SplitUTF8(StringPiece text, const string& delim,
+                 std::vector<string>* result) {
+  // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
+  //   1     0xxxxxxx
+  //   2     110xxxxx  10xxxxxx
+  //   3     1110xxxx  10xxxxxx  10xxxxxx
+  //   4     11110xxx  10xxxxxx  10xxxxxx  10xxxxxx
+  result->clear();
+  result->reserve(text.size());
+  size_t off = 0, len = 0;
+  bool advance = true;
+  for (size_t i = 0; i < text.size(); ++i) {
+    if (off == i) {
+      if ((text[i] & 0x80) == 0x00) {
+        string entry = text.substr(off, 1).ToString();
+        if (delim == "") {
+          // If delim is "" then always advance
+          result->emplace_back(entry);
+        } else if (delim != entry) {
+          // If delim is not the current char, then advance only once
+          if (advance) {
+            advance = false;
+            result->emplace_back(entry);
+          } else {
+            (*result)[result->size() - 1] =
+                (*result)[result->size() - 1] + entry;
+          }
+        } else {
+          // If delim is the current char, then set the advance but
+          // not append the string.
+          advance = true;
+        }
+        off++;
+      } else if ((text[i] & 0xE0) == 0xC0) {
+        len = 2;
+      } else if ((text[i] & 0xF0) == 0xE0) {
+        len = 3;
+      } else if ((text[i] & 0xF8) == 0xF0) {
+        len = 4;
+      } else {
+        result->clear();
+        return errors::InvalidArgument("Invalid UTF8 encoding at position of ",
```

Comment:
Replace "position" with "byte" in the error messages to make it clear that the position refers to the byte and not the character?

Reply:
Done.